docs/jax-101/05.1-pytrees.md

```diff
@@ -187,6 +187,22 @@ plt.scatter(xs, forward(params, xs), label='Model prediction')
 plt.legend();
 ```
 
+## Key paths
+
+In a nested pytree, each leaf has a unique "path of keys" corresponding to its place in the tree. JAX allows you to access the key paths and use them when handling a pytree. The corresponding APIs are:
+
+* `jax.tree_util.tree_flatten_with_path`
+* `jax.tree_util.tree_map_with_path`
```

Comment:
Should we add `keystr` here? We can also add an sentence that says: all we guarantee you can do with a general key path is apply `keystr` to it to get a string.

Also, can we add type annotations to these functions, if they're not too confusing? I think our doc had some.

Reply:
Added - I didn't put type annotation in the doc to preserve the readability, but I added the docstring links (which contains type annotation and more) for each API calls.

Reply:
Good call!